src/algebra/order/monoid.lean

```diff
@@ -110,11 +110,32 @@ class linear_ordered_add_comm_monoid (α : Type*)
 class linear_ordered_comm_monoid (α : Type*)
   extends linear_order α, ordered_comm_monoid α.
 
+/-- Typeclass for expressing that the `0` of a type is less or equal to its `1`. -/
+class zero_le_one_class (α : Type*) [has_zero α] [has_one α] [has_le α] :=
+(zero_le_one : (0 : α) ≤ 1)
+
+@[simp] lemma zero_le_one [has_le α] [has_zero α] [has_one α] [zero_le_one_class α] : (0 : α) ≤ 1 :=
+zero_le_one_class.zero_le_one
+
+lemma zero_le_two [preorder α] [has_one α] [add_zero_class α] [zero_le_one_class α]
+  [covariant_class α α (+) (≤)] : (0 : α) ≤ 2 :=
+add_nonneg zero_le_one zero_le_one
+
+lemma one_le_two [has_le α] [has_one α] [add_zero_class α] [zero_le_one_class α]
+  [covariant_class α α (swap (+)) (≤)] : (1 : α) ≤ 2 :=
+calc 1 = 0 + 1 : (zero_add 1).symm
+   ... ≤ 1 + 1 : add_le_add_right zero_le_one _
```

Comment:
```suggestion
  [covariant_class α α (+) (≤)] : (1 : α) ≤ 2 :=
calc 1 = 1 + 0 : (add_zero _).symm
   ... ≤ 1 + 1 : add_le_add_left zero_le_one _
```
This also works, if you want to maintain the same "non-`swap`" assumption on `covariant_class`.

Reply:
I think these can be `zero_le_two` and `zero_le_two'` in some order. That way we're able to use this theorem in even more general situations.